packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -139,7 +139,34 @@ function useConfigLoader(
     };
   }
 
-  const configReader = ConfigReader.fromConfigs(config.value ?? []);
+  let configReader = ConfigReader.fromConfigs(config.value ?? []);
+
+  const getFullBackendUrl = () => {
+    // Backend.baseUrl should always be defined.
+    let url;
+    try {
+      url = new URL(
+        configReader.getString('backend.baseUrl'),
+        document.location.origin,
+      ).href.replace(/\/$/, '');
+    } catch (err) {
+      // Backend.baseUrl was not a valid URL. This should be caught during the build process.
+    }
+    return url;
+  };
+
+  const relativeBackendConfig = {
+    data: {
+      backend: {
+        baseUrl: getFullBackendUrl(),
+      },
+    },
+    context: 'relative-override',
+  };
+
+  configReader = ConfigReader.fromConfigs(
+    config.value ? [...config.value, relativeBackendConfig] : [],
```

Comment:
I'm thinking that we should only do this if the backend baseUrl is relative. What do you think about adding a check to the CLI that makes sure that the cli config path always starts with a `'/'`? Then we can have a simple `.startsWith('/')` here to figure out whether it's a relative URL or not. If we in the future want to allow for other types of relative URLs we can do that too, but by being strict now we avoid breaking changes in the future.

Reply:
Makes sense, I'll update it.

I had a quick question related to that, in what case would https://github.com/sennyeya/backstage/blob/55817286d4691a2e7b58dbaab4d9a6dfdc1fad49/packages/core-app-api/src/app/AppManager.tsx#L142 have `config.value` be undefined? I want to make sure I'm handling an undefined config correctly.

Reply:
I believe it's for tests, so that the config is immediately empty and we don't have a bunch of rerendering happening in tests while config is loaded. If it's empty I think we'll want an early return with an empty reader.